app/components/annotation/Article.js

```diff
@@ -17,14 +59,45 @@ export default React.createClass({
     let article = this.props.article;
     let topic = this.props.topics[topic_id];
 
+    var text = this.props.article.text;
+    var highlights = this.props.highlights || [];
+
+    var start = 0;
+    var tail = '';
+    var l = highlights.length;
+
+    if (l === 0) {
+      tail = text;
+    } else if (highlights[l - 1].end !== text.length) {
+      tail = <span>{text.substring(highlights[l - 1].end, text.length)}</span>;
+    }
+
     return (
       <div>
         <div className='tua__header-text'>
           Focus on the bold text about '{topic.name}' and answer the questions.
         </div>
-        <div className='article'>{article.text}</div>
+        <div ref={(ref) => this.articleRef = ref} className='article' onClick={this.handleClick}>
+          {Array(highlights.length * 2).fill().map((_,i) => {
```

Comment:
is `_` standing in for `highlight` here? if so i'd just name it `highlight` for clarity.